test/integration/shopping-cart.repository.integration.ts

```diff
@@ -0,0 +1,101 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/example-shopping
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {ShoppingCartRepository} from '../../src/repositories/shopping-cart.repository';
+import {ShoppingCart} from '../../src/models';
+import {expect} from '@loopback/testlab';
+import Docker = require('dockerode');
+
+describe('ShoppingCart KeyValue Repository', () => {
+  let repo: ShoppingCartRepository;
+  let cart1: ShoppingCart;
+  let cart2: ShoppingCart;
+  let docker: Docker;
+  let container: Docker.Container;
+
+  before(async function() {
+    // tslint:disable-next-line:no-invalid-this
+    this.timeout(0);
+    docker = new Docker();
+    container = await docker.createContainer({
+      Image: 'redis',
+      Tty: true,
+      ExposedPorts: {
+        '6379/tcp': {},
+      },
+      HostConfig: {
+        PortBindings: {
+          '6379/tcp': [
+            {
+              HostPort: '6379',
+            },
+          ],
+        },
+      },
+    });
+    await container.start();
+  });
+
+  after(async () => {
+    await container.stop();
+    await container.remove();
+  });
+
+  before(() => {
+    cart1 = new ShoppingCart({
+      userId: 'u01',
+      items: [
+        {
+          productId: 'p1',
```

Comment:
Please follow our best practices, in particular [Use data builders](https://loopback.io/doc/en/lb4/Testing-your-application.html#use-test-data-builders).